### PR TITLE
Integrate console runner with IJ plugin

### PR DESCRIPTION
**Pull request by raniejade**
With this you can now run, Spek 2.x specs within IntelliJ.

### CODE REVIEW

File: spek-ide-plugin/intellij-jvm/src/main/kotlin/org/spekframework/intellij/spek2.kt [hunk 1]
```diff
@@ -86,7 +88,52 @@ open class Spek2JvmRunConfiguration(name: String,
...
+                    myConfiguration.alternativeJrePath
+                } else {
+                    null
+                }
+
+                // AndroidJUnitPatcher
+                val pathType = JavaParameters.JDK_AND_CLASSES_AND_TESTS
+                JavaParametersUtil.configureModule(module, params, pathType, jreHome)
+
+                val jars = mutableListOf(
```
Comment (artem-zinnatullin) [inline spek-ide-plugin/intellij-jvm/src/main/kotlin/org/spekframework/intellij/spek2.kt]: Can be immutable as far as i see

File: spek-ide-plugin/interop-jvm/src/main/kotlin/org/spekframework/ide/ServiceMessageAdapter.kt [hunk 2]
```diff
@@ -4,26 +4,32 @@ import org.spekframework.spek2.runtime.execution.ExecutionListener
...
+    }
 
     override fun testExecutionStart(test: TestScopeImpl) {
+        durations[test.path] = System.currentTimeMillis()
         out("testStarted name='${test.path.name.toServiceMessageSafeString()}'")
     }
 
     override fun testExecutionFinish(test: TestScopeImpl, result: ExecutionResult) {
         val name = test.path.name.toServiceMessageSafeString()
+        val duration = System.currentTimeMillis() - (durations[test.path] ?: 0)
```
Comment (artem-zinnatullin) [inline spek-ide-plugin/interop-jvm/src/main/kotlin/org/spekframework/ide/ServiceMessageAdapter.kt]: Let's fail if `duration[test.path]` is null? Seems impossible thus no need to write defensive code

File: spek-runtime/common/src/main/kotlin/org/spekframework/spek2/runtime/scope/ScopeImpl.kt [hunk 3]
```diff
@@ -33,12 +33,17 @@ open class GroupScopeImpl(id: ScopeId,
...
         children.filter { it.path.isRelated(path) }
             .forEach {
                 if (it is GroupScopeImpl) {
                     it.filterBy(path)
                 }
+                tmp.add(it)
             }
+
+        children.clear()
+        children.addAll(tmp)
```
Comment (artem-zinnatullin) [inline spek-runtime/common/src/main/kotlin/org/spekframework/spek2/runtime/scope/ScopeImpl.kt:46]: Cover with tests?

**Comment by raniejade** [comment]
@artem-zinnatullin changes in! Thanks for the prompt review!

File: spek-runtime/jvm/src/test/kotlin/org/spekframework/spek2/jvm/FilterTest.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.runtime.scope.PathBuilder
+import org.spekframework.spek2.runtime.test.AbstractSpekRuntimeTest
+
+class FilterTest: AbstractSpekRuntimeTest() {
+    @Test
+    fun filter() {
+        class TestSpek: Spek({
+            group("group") {
+                test("test") {}
```
Comment (artem-zinnatullin) [inline spek-runtime/jvm/src/test/kotlin/org/spekframework/spek2/jvm/FilterTest.kt]: Let's add some logic, like counter increment to verify that correct test was selected and executed?

**Review by artem-zinnatullin — APPROVED** [review]
Looks good, one comment :)

**Pull request merged**